agent/src/main/java/com/code_intelligence/jazzer/autofuzz/Meta.java

```diff
@@ -161,46 +207,112 @@ public static Object autofuzz(FuzzedDataProvider data, Method method, Object thi
   }
 
   public static Object consume(FuzzedDataProvider data, Class<?> type) {
+    return consume(data, type, null);
+  }
+
+  static Object consume(FuzzedDataProvider data, Class<?> type, AutofuzzCodegenVisitor visitor) {
     if (type == byte.class || type == Byte.class) {
-      return data.consumeByte();
+      byte result = data.consumeByte();
+      if (visitor != null)
```

Comment:
Do you have an idea about the performance overhead caused by the if statements when `visitor` is `null`?

Reply:
I expect it to be negligible for two reasons:
1. The entire method consists of a chain of `if` statements and these calls only every add a constant number per `consume` call. Most branches do far more work than this `if`.
2. Since the branch structure is pretty simple and it is a simple `null` check, I would expect this to JIT compile down to a single compare instruction that is never taken, which should be virtually free due to branch prediction.

Once Autofuzz has become a bit more mature, we should certainly profile it thoroughly though and revisit these points.

Reply:
OK, then let's revisit this point later